### PR TITLE
clear local storage

### PR DESCRIPTION
**Pull request by hutnikau**

### CODE REVIEW

File: views/js/testRunner/testMetaData.js [hunk 1]
```diff
@@ -162,7 +162,27 @@ define([
...
+                            removed++;
+                        }
+                    }
+                    if (removed) {
+                        setLocalStorageData(val);
+                    } else {
+                        throw domException;
+                    }
+                }
+            }
```
Comment (krampstudio) [inline views/js/testRunner/testMetaData.js]: you should aslo re-throw it if it is another type of exception (for example if the value isn't valid)

Comment (hutnikau) [inline views/js/testRunner/testMetaData.js]: Done

**Comment by krampstudio** [comment]
I'll reassign it to @jsconan for the testing. I'm agree with the implementation (at some point we will need a util module for storage that handle those edge cases, but for now it's sufficient).

**Comment by jsconan** [comment]
Test OK for the immediate behavior: the localStorage is well cleaned at each transition.
And there is no more javascript error on tear down.

But, unfortunately the clean up does not work well when the storage is out of space.
Not all the keys are removed due to a misconception of your erase loop. By example you can read this article for some explanation about iterate and remove: http://billgonemad.com/blog/?p=34

Take also into account that erasing will take a while when the storage is full, and the browser will hang during this process.

Here is a more consistent erase loop:

```javascript
function cleanStorage(currentKey) {
    var key,
        removed = 0,
        i = localStorage.length;
    while (i) {
        key = localStorage.key(i--);
        if (/^testMetaData_.*/.test(key) && key !== currentKey) {
            window.localStorage.removeItem(key);
            removed++;
        }
    }
    return removed;
}
```

**Comment by hutnikau** [comment]
@jsconan i just updated the code. Please review it again.

**Comment by jsconan** [comment]
Ok, all is working.
